### ISSUE TITLE
git image doesn't follow semver

### ISSUE THREAD

**Issue by wlynch** [chainguard-images/images#406]
### Which image/versions are related to this issue/feature request?

git

### Issue/Feature description

From https://github.com/chainguard-images/images/blob/main/BEST_PRACTICES.md#tagging

> Note that version tags come before variant tags. So it’s acme:latest-dev and acme:1.2.3-nonroot, not acme:dev-latest etc.

The git image doesn't follow this rule - it uses `git:root-1.2.3`. I stumbled on this because I was trying to set up dependabot to auto-update a git image, but it only seemed to work for [image tags that followed semver](https://github.com/wlynch/test/pull/22) formatting.

If we swapped the order here I think things would work as expected!

cc @imjasonh @jdolitsky

**Comment by patflynn** [comment]
@imjasonh @wlynch can we close this?

**Comment by imjasonh** [comment]
No, we're still pushing the root image as `root-1.2.3` instead of the documented `1.2.3-root`, because rootiness isn't a variant, it's a separate image config.

#405 added a `-dev` variant that makes `1.2.3-dev` available, but that image still doesn't run as root.
